apps/meteor/tests/e2e/page-objects/marketplace.ts

```diff
@@ -0,0 +1,45 @@
+import type { Locator, Page } from '@playwright/test';
+
+export class Marketplace {
+	private readonly page: Page;
+
+	constructor(page: Page) {
+		this.page = page;
+	}
+
+	get btnUploadPrivateApp(): Locator {
+		return this.page.locator('[data-qa-id="upload-private-app"]');
+	}
+
+	get btnInstallPrivateApp(): Locator {
+		return this.page.locator('[data-qa-id="install-private-app"]');
+	}
+
+	get btnUploadPrivateAppFile(): Locator {
+		return this.page.locator('role=button[name="Browse Files"]');
+	}
+
+	get appStatusTag(): Locator {
+		return this.page.locator('[data-qa-type="app-status-tag"]');
+	}
+
+	get confirmAppUploadModalTitle(): Locator {
+		return this.page.locator('[data-qa-id="confirm-app-upload-modal-title"]');
+	}
+
+	get confirmAppUploadModalButton(): Locator {
+		return this.page.locator('[data-qa-id="confirm-app-upload-modal-button"]');
+	}
```

Comment:
For consistency I'd recommend starting this getter with `btn` (e.g `btnConfirmAppUploadModal` ). Also, if possible I recommend using a combination of role and name to find this element.